### PR TITLE
Add constructor to support passing static dimensions in TDQ

### PR DESCRIPTION
**Pull request by bharatmotwani**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

**Comment by cdeszaq** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- fili-core/src/main/java/com/yahoo/bard/webservice/data/DruidQueryBuilder.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/Fili/fili/pullRequest?prid=5561291&bid=18237760)

File: CHANGELOG.md [hunk 1]
```diff
@@ -9,6 +9,9 @@ Current
 -------
 ### Fixed:
 
+- [Fix: Add withDimension in TemplateDruidQuery](https://github.com/yahoo/fili/pull/1058)
+    * Ability to add dimension in TemplateDruidQuery on the fly.
```
Comment (michael-mclawhorn) [inline CHANGELOG.md:13]: Modify a TDQ to add a dependent dimension.

File: fili-core/src/test/groovy/com/yahoo/bard/webservice/data/metric/TemplateDruidQuerySpec.groovy [hunk 2]
```diff
@@ -469,4 +470,45 @@ class TemplateDruidQuerySpec extends Specification {
                 .find { it -> it.getName() == arithmeticPostAggFromAliasName}
         resultArithmeticPa.getPostAggregations() == [arithmeticPostAggOperand1, resultAliasPa] as List
     }
+
+    def "Simple tests by passing static dimensions while building TDQ"() {
```
Comment (michael-mclawhorn) [inline fili-core/src/test/groovy/com/yahoo/bard/webservice/data/metric/TemplateDruidQuerySpec.groovy:474]: test name should somewhat state what the outcome is.  
So something like 'Dimensions are retained when tdqs are merged'
Also, I'd suggest doing this with a where block where you test the following cases:
leftside = [dim1], rightside=[]
leftside=[dim1], rightside[dim2]
leftside=[dim1,dim2], rightside[dim1]
Just to make sure merging does what we want it to do with one sideded and overlapping dimension sets.

Comment (michael-mclawhorn) [reply fili-core/src/test/groovy/com/yahoo/bard/webservice/data/metric/TemplateDruidQuerySpec.groovy:474]: If you do do 'where' then the you should put variables in the method name with the list of dimensions being passed, e.g. "When #dimensions1 merged with #dimension2 the resulting dimension set is #expected"

File: fili-core/src/test/groovy/com/yahoo/bard/webservice/druid/serializers/LookupDimensionToDimensionSpecSpec.groovy [hunk 3]
```diff
@@ -67,6 +67,7 @@ class LookupDimensionToDimensionSpecSpec extends Specification{
     def "Given lookup dimension with no namespace serialize using dimension serializer"() {
         given:
         apiRequest.getDimensions() >> ([resources.d10])
```
Comment (michael-mclawhorn) [inline fili-core/src/test/groovy/com/yahoo/bard/webservice/druid/serializers/LookupDimensionToDimensionSpecSpec.groovy:69]: Can these be safely removed or not?  (applies throughout)

Comment (bharatmotwani) [reply fili-core/src/test/groovy/com/yahoo/bard/webservice/druid/serializers/LookupDimensionToDimensionSpecSpec.groovy:69]: No , it fails in constraints code with null pointer.
https://github.com/yahoo/fili/blob/master/fili-core/src/main/java/com/yahoo/bard/webservice/table/resolver/BaseDataSourceConstraint.java#L44

**Pull request merged**